src/content/reference/react/cache.md

```diff
@@ -0,0 +1,306 @@
+
+</Pitfall>
+
+### Preload data {/*preload-data*/}
+
+By wrapping a long-running data fetch in `cache`, you can kick off the asynchronous work prior to rendering the component.
+
+```jsx
+const getUser = cache(async (id) => {
+  return await db.user.query(id);
+}
+
+async function Profile({id}) {
+  const user = await getUser(id);
+  // ...
+}
+
+function Page({id}) {
+  getUser(id);
+  // ... some computational work
+  return (
+    <>
+	    ...
+      <Profile id={id} />
+    </>
+  );
+}
+```
+
+When rendering `Page`, the component calls `getUser` but note that it doesn't use the returned data. This early `getUser` call kicks off the asynchronous database query that occurs while `Page` is doing other computational work and rendering children. 
+
+When rendering `Profile`, we call `getUser` again. In the ideal case, the initial `getUser` call has already returned and cached the user data. So when `Profile` asks and waits for this data, it can simply read from the cache without requiring another remote procedure call. Even if the data request hasn't been completed, preloading data in this pattern reduces delay in data-fetching..
+
+<Pitfall>
+
+Memoization only occurs within component renders.
+
+```jsx
+import {cache} from 'react';
+
+const getUser = cache(async (userId) => {
+  return await db.user.query(userId);    
+});
+
+// This call to `getUser` will not update the cache.
+getUser('demo-id');
+
+async function DemoProfile() {
+  // This `getUser` will have a cache miss
+  const user = await getUser('demo-id');
+  return (<Profile user={user} />);
+}
+```
+
+You may attempt to preload some work outside of a component. React only provides cache access to the memoizing function in a component render. 
+</Pitfall>
+
+<DeepDive>
+
+#### What's the difference between `cache`, [`memo`](/reference/react/memo), and [`useMemo`](/reference/react/useMemo)? {/*cache-memo-usememo*/}
+All mentioned APIs offer memoization but the difference is what they're intended to memoize, who can access the cache, and when their cache is invalidated.
+
+[`useMemo`](/reference/react/useMemo) is a hook that is used in a client component or another hook. `useMemo` takes a function and a list of dependencies. During the component render, `useMemo` will only call the function when the dependencies have changed and will cache the result. 
+
+You can think of `useMemo`'s cache as a size of one, where it can only cache the last work it did based on the last value of its dependencies. Only the component instance can benefit from this cache to skip work across re-renders. If you rendered two instances of the same component, the `useMemo` of one wouldn't help the other to skip work. The cache is invalidated each time the dependencies change or the component is unmounted.
+
+```jsx
+'use client';
+
+function Weather(city) {
+    // assume `getTemp` is an expensive computation
+	const temp = useMemo(() => getTemp(city)), city); 
```

Comment:
Nit: `getTemp` sounds like a data fetch, which isn't allowed in `useMemo`. Maybe `calculateTemp` instead, or something more realistic like `filterTodos`.

Reply:
Maybe the issue here is that we're comparing `cache` with `useMemo` when it's really more like `useCallback`.

Reply:
I think the memoized function returned from `cache` is more like `useMemo`. It is caching the returned value of the function. I can update the example